lib/evmone/execution_state.hpp

```diff
@@ -27,20 +28,48 @@ using intx::uint256;
 /// Provides memory for EVM stack.
 class StackSpace
 {
+    static uint256* allocate() noexcept
+    {
+        static constexpr auto alignment = sizeof(uint256);
+        static constexpr auto size = limit * sizeof(uint256);
+#ifdef _MSC_VER
+        // MSVC doesn't support aligned_alloc() but _aligned_malloc() can be used instead.
+        const auto p = _aligned_malloc(size, alignment);
+#else
+        const auto p = std::aligned_alloc(alignment, size);
+#endif
+        return static_cast<uint256*>(p);
+    }
+
+    struct Deleter
+    {
+        // TODO(C++23): static
+        void operator()(void* p) noexcept
+        {
+#ifdef _MSC_VER
+            // For MSVC the _aligned_malloc() must be paired with _aligned_free().
+            _aligned_free(p);
+#else
+            std::free(p);
+#endif
+        }
+    };
+
+    /// The storage allocated for maximum possible number of items.
+    /// Items are aligned to 256 bits for better packing in cache lines.
+    std::unique_ptr<uint256, Deleter> m_stack_space;
+
 public:
     /// The maximum number of EVM stack items.
     static constexpr auto limit = 1024;
 
+    StackSpace() noexcept : m_stack_space{allocate()} {}
```

Comment:
I wonder if 
```cpp
std::unique_ptr<uint256[]>(new (std::align_val_t(sizeof(uint256)) uint256[limit]);
``` 
achieves the same.

Reply:
It looks like, however this specifies the array element alignment, so I wonder what happens if you specify 64 for uint256. I plan to align the allocation to `2048*32`.